EIPS/eip-5750.md

```diff
@@ -8,103 +8,91 @@ status: Review
 
-1. Any methods complying with this EIP, such as overloaded `transfer` and `vote` could use string reasons as the extra data.
-2. Existing EIPs that have exported methods compliant with this EIP can be extended for behaviors such as using the extra data to prove endorsement or as a salt, nonce, or commitments for reveal/commit schemes.
-3. Data can be passed forward to callbacks.
+There are two ways to achieve extensibility for existing functions. Each comes with their set of challenges:
 
-### Why there needs to be a standard for these use case
+1. Add a new method
+  * What will the method name be?
+  * What will the parameters be?
+  * How many use-cases does a given method signature support?
+  * Does this support off-chain signatures?
+2. Use one or more existing parameters, or add one or more new ones
+  * Should existing parameters be repurposed, or should more be added?
+  * How many parameters should be used?
+  * What are their sizes and types?
 
-First of all, there are multiple ways to achieve extensibility for existing functions.
+Standardizing how methods can be extended helps to answer these questions.
 
-1. _Add a new method_, then these questions need to be answered:
-
-- what will be the method name?
-- what will be the full list of parameters?
-- How many use-case do a given method name support?
-- Do/Will they support off-chain signatures?
-
-2. _Use an existing parameter(s)_: then these questions need to be answered:
-
-- will it be one parameter?
-- w- will it be multiple parameters?
-- what are their sizes and types?
-
-Standardizing how method will be extended helps answer this questions.
-
-In particular, this EIP aims at achieving maximum backward (e.g. [EIP-721](./eip-721.md), [EIP-1155](./eip-1155.md)) and future compatibility e.g. supporting commit-reveal feature for arbitrary method ([EIP-5732](./eip-5732.md)), or adding digital signatures alongside with a method call. The goal and motivation is to allow other implementers and EIPs to further depend on the compatibility based on the specifications this EIP so all compliant method interface automatically becomes eligible for future new behaviors.
+Finally, this EIP aims to achievie maximum backward and future compatibility. Many EIPs already partially support this EIP, such as [EIP-721](./eip-721.md) and [EIP-1155]. This EIP supports many use cases, from commit-reveal schemes ([EIP-5732](./eip-5732.md)), to adding digital signatures alongside with a method call. Other implementers and EIPs should be able to depend on the compatibility granted by this EIP so that all compliant method interfaces are eligible for future new behaviors.
```

Comment:
```suggestion
Finally, this EIP aims to achievie maximum backward and future compatibility. Many EIPs already partially support this EIP, such as [EIP-721](./eip-721.md) and [EIP-1155](./eip-1155.md). This EIP supports many use cases, from commit-reveal schemes ([EIP-5732](./eip-5732.md)), to adding digital signatures alongside with a method call. Other implementers and EIPs should be able to depend on the compatibility granted by this EIP so that all compliant method interfaces are eligible for future new behaviors.
```